internal/cost-lru/cost_lru.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package costlru
```

Comment:
Why not to put it into `/lib/utils/lru-cache` along with our old LRU cache impl?

Reply:
I wasn't positive I wanted this to be importable from outside the gossamer repo.